src/code.rs

```diff
@@ -555,6 +557,66 @@ fn build_table_fns(
     }}
 "##));
 
+    #[cfg(feature = "advanced-queries")]
+    // Table::filter() helper fn
+    {
+        let diesel_backend = &config.diesel_backend;
+        let filters = table
+            .columns
+            .iter()
+            .map(|column| {
+                let column_name = column.name.to_string();
+
+                if column.is_nullable {
+                    format!(
+                        r##"
+            if let Some(filter_{column_name}) = filter.{column_name}.clone() {{
+                query = if filter_{column_name}.is_some() {{ 
+                    query.filter({schema_path}{table_name}::{column_name}.eq(filter_{column_name}))
+                }} else {{
+                    query.filter({schema_path}{table_name}::{column_name}.is_null())
+                }};
+            }}"##
+                    )
+                } else {
+                    format!(
+                        r##"
+            if let Some(filter_{column_name}) = filter.{column_name}.clone() {{
+                query = query.filter({schema_path}{table_name}::{column_name}.eq(filter_{column_name}));
+            }}"##
```

Comment:
do we really need to `.clone` everything? shouldnt it be possible to just use [partial-moves](https://doc.rust-lang.org/rust-by-example/scope/move/partial_move.html) because the function takes in full ownership of the passed filter instead of a reference and does not return the filter itself?

Reply:
Thanks for pointing this out -- I'll have to look into it later

Reply:
https://github.com/Wulf/dsync/issues/130